registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/UserNotificationApi.java

```diff
@@ -0,0 +1,316 @@
+	private Integer workerPoolSize;
+
+	@Value("${mosip.regproc.user.notification.api.eventbus.port}")
+	private String eventBusPort;
+
+	@Value("${mosip.registration.processor.datetime.pattern}")
+	private String dateTimePattern;
+
+	@Value("${mosip.regproc.user.notification.api.id}")
+	private String id;
+
+	@Value("${mosip.regproc.user.notification.api.version}")
+	private String version;
+	
+	private static final String STAGE_PROPERTY_PREFIX = "mosip.regproc.reprocessor.";
+
+	@Autowired
+	MosipRouter router;
+
+	@Value("${server.servlet.path}")
+	private String contextPath;
+
+	private MosipEventBus mosipEventBus = null;
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(UserNotificationApi.class);
+
+	/** The registration status service. */
+	@Autowired
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	/** The module id. */
+	public static String MODULE_ID = PlatformSuccessMessages.RPR_USER_NOTIFICATION_API_SUCCESS.getCode();
+
+	/** The module name. */
+	public static String MODULE_NAME = ModuleName.USER_NOTIFICATION_API.toString();
+
+	/** The core audit request builder. */
+	@Autowired
+	AuditLogRequestBuilder auditLogRequestBuilder;
+
+	@Autowired
+	UserNotificationRequestValidator UserNotificationRequestValidator;
+
+	/**
+	 * Deploy verticle.
+	 */
+	public void deployVerticle() {
+		mosipEventBus = this.getEventBus(this, clusterManagerUrl, workerPoolSize);
+	}
+
+	@Override
+	public void start() {
+		router.setRoute(this.postUrl(getVertx(), null, null));
+		this.routes(router);
+		this.createServer(router.getRouter(), Integer.parseInt(port));
+	}
+
+	@Override
+	public Integer getEventBusPort() {
+		return Integer.parseInt(eventBusPort);
+	}
+
+	/**
+	 * contains all the routes in this stage
+	 *
+	 * @param router
+	 */
+	private void routes(MosipRouter router) {
+		router.post(contextPath + "/userNotification");
+		router.handler(this::processURL, this::failure);
+	}
+
+	/**
+	 * method to process the context received.
+	 *
+	 * @param ctx the ctx
+	 */
+	public void processURL(RoutingContext ctx) {
+		LogDescription description = new LogDescription();
+		InternalRegistrationStatusDto registrationStatusDto;
+		JsonObject obj = ctx.getBodyAsJson();
+		String rid = null;
+		try {
+			UserNotificationDTO UserNotificationDTO = JsonUtil.readValueWithUnknownProperties(obj.toString(),
+					UserNotificationDTO.class);
+			UserNotificationRequestDTO UserNotificationRequestDTO = UserNotificationDTO.getRequest();
+			rid = UserNotificationRequestDTO.getRid();
+			regProcLogger.debug("UserNotificationApi:processURL called for registration id {}",
+					UserNotificationDTO.getRequest().getRid());
+			UserNotificationRequestValidator.validate(UserNotificationDTO);
+			String user = getUser(ctx);
+
+			registrationStatusDto = registrationStatusService.getRegistrationStatus(rid,UserNotificationRequestDTO.getRegType(),1,null);
```

Comment:
Why is 1 hardcoded here ?

Reply:
I don't find any response to this

Reply:
it is not being used in dao layer to fetch data. in all other cases it is a part of message DTO. so message DTO is not being used here so kept as 1.

Reply:
We have to check the behaviour of iteration and design this accordingly so it make sense in the user notification feature. As per the code it takes the 0th element without order from the list and uses that to send the notification. 
Please discuss with @sowmya695